https/fetch.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package https
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Request encapsulates an HTTPs request.
+type Request struct {
+	// URL is the HTTPs endpoint.
+	URL string
+	// Method is the HTTP method to use in the request.
+	Method string
+	// TrustedCertFingerprint is the sha256 hash of a server's trusted
+	// (self-signed) TLS certificate.
+	TrustedCertFingerprint []byte
+}
+
+// Response encapsulates an HTTPs response.
+type Response struct {
+	// Data is the received request payload.
+	Data []byte
+	// HTTPStatusCode is the HTTP status code of the response.
+	HTTPStatusCode int
+	// RedirectURL is the Location header of a HTTP redirect response.
+	RedirectURL string
```

Comment:
Perhaps you return a list of headers and the TS code can process the location.

Reply:
If it's okay, I'd rather stick to the bare minimum to avoid having to serialize more complex types, i.e. list/map.